client/my-sites/sharing/connections/services-group.jsx

```diff
@@ -1,123 +1,87 @@
-			if ( 'eventbrite' === service.name && ! utils.userCan( 'manage_options', site ) ) {
-				return false;
-			}
-
-			// Omit if Publicize service and user cannot publish
-			if ( 'publicize' === service.type && ! utils.userCan( 'publish_posts', site ) ) {
-				return false;
-			}
-
-			return true;
-		} );
-	},
-
-	renderService: function( service ) {
+const NUMBER_OF_PLACEHOLDERS = 4;
+
+class SharingServicesGroup extends Component {
+
+	static propTypes = {
+		site: PropTypes.object,
+		user: PropTypes.object,
+		connections: PropTypes.object,
+		onAddConnection: PropTypes.func,
+		onRemoveConnection: PropTypes.func,
+		onRefreshConnection: PropTypes.func,
+		onToggleSitewideConnection: PropTypes.func,
+		initialized: PropTypes.bool,
+		services: PropTypes.array,
+		title: PropTypes.string.isRequired,
+		description: PropTypes.string
+	};
+
+	static defaultProps = {
+		onAddConnection: function() {},
+		onRemoveConnection: function() {},
+		onRefreshConnection: function() {},
+		onToggleSitewideConnection: function() {},
+		initialized: false
+	};
+
+	renderService( service ) {
 		return <Service
-			key={ service.name }
-			site={ this.props.site }
-			user={ this.props.user }
+			key={ service.ID }
 			service={ service }
-			connections={ this.props.connections }
-			onAddConnection={ this.props.onAddConnection }
-			onRemoveConnection={ this.props.onRemoveConnection }
-			onRefreshConnection={ this.props.onRefreshConnection }
-			onToggleSitewideConnection={ this.props.onToggleSitewideConnection } />;
-	},
+			{ ...this.props } />;
+	}
 
-	renderServicePlaceholders: function() {
-		// The Array constructor isn't used here because constructed arrays
-		// can't be mapped since it doesn't truly contain any values
-		return Array.apply( null, Array( NUMBER_OF_PLACEHOLDERS ) ).map( function( value, i ) {
-			return <ServicePlaceholder key={ 'service-placeholder-' + i } />;
-		} );
-	},
+	renderServicePlaceholders() {
+		return times( NUMBER_OF_PLACEHOLDERS, ( index ) => <ServicePlaceholder key={ 'service-placeholder-' + index } /> );
+	}
 
-	renderServices: function( services ) {
+	renderServices() {
 		if ( this.props.initialized ) {
-			return services.map( this.renderService, this );
+			return this.props.services.map( this.renderService, this );
 		}
 		return this.renderServicePlaceholders();
-	},
+	}
 
-	render: function() {
-		var services = this.getEligibleServices(),
-			classes = classNames( 'sharing-services-group', {
-				'is-empty': this.props.initialized && ! services.length
-			} );
+	render() {
+		const classes = classNames( 'sharing-services-group', {
+			'is-empty': this.props.initialized && ! this.props.services.length
+		} );
 
 		return (
 			<div className={ classes }>
+				<QueryKeyringServices />
 				<SectionHeader label={ this.props.title } />
 				<ul className="sharing-services-group__services">
-					{ this.renderServices( services ) }
+					{ this.renderServices() }
```

Comment:
A point of future refactoring, I'd like to eliminate the `renderX` functions in favor of either inlining them into the `render` or creating separate components. Would probably not be too difficult to inline here.